### DISCUSSION TITLE
Tipps for Reverse Engineering Spotify

### DISCUSSION THREAD

**Discussion by Stefan-WW** [librespot-org/librespot#1323] [General]

**Comment by roderickvd** [comment]
If you have to protobuf binary from the request / response, then you could feed it to protoc and at least get the fields and data types. Then, if you know from the app what data it should represent (e.g. podcast id, number of listens, date/time) then you can start guessing which field is which and make your own protobuf definition.

**Comment by Mationsills39** [comment]
I had a similar problem with extracting data from a mobile app a while back. It took me ages to find the right tools that could handle protobuf files. One trick that helped was comparing the extracted files from the desktop and mobile versions. I noticed subtle differences in the request headers. It might sound tedious, but sometimes trying out different decoding tools can reveal hidden patterns. For engineering projects, I used [https://essays.edubirdie.com/engineering-assignment-help](https://essays.edubirdie.com/engineering-assignment-help) for engineering assignment help, which saved me loads of time debugging issues. Keep experimenting!
